test/common_checks.py

```
@@ -0,0 +1,334 @@
+        assert obj_type in info_to_type(obj)
+    if obj_description is not None and not is_fake_set:
+        assert "description" in info_to_usermeta(obj)
+        assert info_to_usermeta(obj)["description"] == obj_description
+    if n_items is not None and not is_fake_set:
+        assert "item_count" in info_to_usermeta(obj)
+        assert info_to_usermeta(obj)["item_count"] == str(n_items)
+
+
+def check_save_set_output(
+    obj: dict[str, Any],
+    set_name: str,
+    set_type: str | None = None,
+    set_description: str | None = None,
+    n_items: int | None = None,
+    **kwargs,
+) -> None:
+    """Check the output of a save_set command.
+
+    The output should be of the form
+    { "set_ref": <upa>, "set_info": <kbase info list> }
+
+    :param obj: API output to check
+    :type obj: dict[str, Any]
+    :param set_name: the set name
+    :type set_name: str
+    :param set_type: the type of the set, defaults to None
+    :type set_type: str | None, optional
+    :param set_description: user-entered description of the set, defaults to None
+    :type set_description: str | None, optional
+    :param n_items: number of items in the set, defaults to None
+    :type n_items: int | None, optional
+    :param kwargs: leftover params - not used
+    :type kwargs: dict[str, Any], optional
+    """
+    assert obj is not None
+    assert "set_ref" in obj
+    assert "set_info" in obj
+    assert obj["set_ref"] == info_to_ref(obj["set_info"])
+    is_info_object(obj["set_info"], set_name, set_type, set_description, n_items)
+
+
+def check_second_item(
+    obj: dict[str, Any],
+    expected_item_data: dict[str, str],
+) -> None:
+    """Check the second item in a set.
+
+    :param obj: API output to check, with items stored in obj["data"]["items"]
+    :type obj: dict[str, Any]
+    :param expected_item_data: expected item data (should have ref and label)
+    :type expected_item_data: dict[str, str]
+    """
+    second_item = obj["data"]["items"][1]
```

Not sure if you want to make sure there's a second set item there to begin with or just let it throw the usual error (IndexError I think).

---

All the sets should have at least two items in them, so if there is an index error, it will indicate that something is wrong elsewhere.